lib/pyfrc/sim/pygame_joysticks.py

```diff
@@ -27,23 +27,19 @@ def initJoystickList(self, joystickList):
     def update(self):
         pygame.event.get()
 
-        for i in range(len(self.joysticks)):
-            joystick = self.joysticks[i]
-            ui_joystick = self.ui.joysticks[i]
+        for joystick, (ui_axes, ui_buttons) in zip(self.joysticks, self.ui.joysticks):
 
-            ui_axes = ui_joystick[0]
             for axis in range(joystick.get_numaxes()):
-                if axis == 6:
+                if axis >= len(ui_axes):
                     break
 
                 ui_current_axis = ui_axes[axis]
 
                 value = joystick.get_axis(axis)
                 ui_current_axis.set_value(value)
 
-            ui_buttons = ui_joystick[1]
             for button in range(joystick.get_numbuttons()):
-                if button == 10:
+                if button > len(ui_buttons):
```

Comment:
Wait, why is the above `>=` but this one strictly `>`? It seems like this should be `>=` as well.

Although... this is probably better spelt `range(min(joystick.get_numbuttons(), len(ui_buttons)))`.